src/setup/App.tsx

```diff
@@ -19,86 +13,105 @@ import { NotFoundPage } from "@/views/notfound/NotFoundView";
 import { V2MigrationView } from "@/views/other/v2Migration";
 import { SearchView } from "@/views/search/SearchView";
 
-function App() {
+// eslint-disable-next-line react/function-component-definition, react/prop-types
+const LegacyUrlView: React.FC = ({ children }) => {
   const location = useLocation();
-  const history = useHistory();
+  const [redirectUrl, setRedirectUrl] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Call the conversion function and set the redirect URL if necessary
+    convertLegacyUrl(location.pathname).then((convertedUrl) => {
+      if (convertedUrl) {
+        setRedirectUrl(convertedUrl);
+      }
+    });
+  }, [location.pathname]);
+
+  if (redirectUrl) {
+    return <Redirect to={redirectUrl} />;
+  }
 
-  // Call the conversion function and redirect if necessary
-  convertLegacyUrl(location.pathname).then((convertedUrl) => {
-    if (convertedUrl) {
-      history.replace(convertedUrl);
-    }
-  });
+  // eslint-disable-next-line react/jsx-no-useless-fragment
+  return <>{children}</>;
```

Comment:
No need for fragment
```suggestion
  return children;
```